addons/hr_expense/models/ir_actions_report.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import io
+from odoo import models
+from odoo.tools import pdf
+from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
+        # OVERRIDE
+        res = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids)
+        if not res_ids:
+            return res
+        report = self._get_report(report_ref)
+        if report.report_name in ('hr_expense.report_expense_sheet'):
+            expense_sheets = self.env['hr.expense.sheet'].browse(res_ids)
+            streams_to_append = {}
+            if expense_sheets:
+                for expense_sheet in expense_sheets:
+                    # Will contains the expense report
+                    stream = res[expense_sheet.id]['stream']
+                    streams_to_append[expense_sheet.id] = stream
+
+                    attachments = self.env['ir.attachment'].search([('res_id', '=', expense_sheet.expense_line_ids.mapped('id')), ('res_model', '=', 'hr.expense')])
+                    expense_report = OdooPdfFileReader(stream, strict=False)
+                    output_pdf = OdooPdfFileWriter()
+                    output_pdf.appendPagesFromReader(expense_report)
+                    for attachment in attachments:
+                        if attachment.mimetype == 'application/pdf':
+                            attachment_stream = pdf.to_pdf_stream(attachment)
+                            attachment_pdf = OdooPdfFileReader(attachment_stream, strict=False)
+                            output_pdf.appendPagesFromReader(attachment_pdf)
+                        else:
+                            data['attachment'] = attachment
+                            attachment_img = self._render_qweb_pdf_prepare_streams('hr_expense.report_expense_sheet_img', data, res_ids=res_ids)
+                            attachment_img_stream = OdooPdfFileReader(attachment_img[expense_sheet.id]['stream'], strict=False)
+                            output_pdf.appendPagesFromReader(attachment_img_stream)
+
+                        new_pdf_stream = io.BytesIO()
+                        output_pdf.write(new_pdf_stream)
+                        res[expense_sheet.id]['stream'] = new_pdf_stream
+                        stream.close()
```

Comment:
Shouldn't this be done one tab left ? At the same level as `stream = ...` (where you get the stream) ?
Maybe this portion is worth a comment to indicate what you're doing. :thinking: